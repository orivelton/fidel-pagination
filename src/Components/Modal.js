import React from 'react';

const Modal = ({ openModal, handleModal, currentItem }) => {
  const {card: { scheme, lastNumbers }} = currentItem;
  const { location: { address, city, countryCode } } = currentItem;
  const { identifiers: { MID }} = currentItem;
  return (
    <div className={openModal ? 'modal is-active' : 'modal'}>
      <div className="modal-background" onClick={handleModal}></div>
      <div className="modal-card">
        <header className="modal-card-head">
          <p className="modal-card-title">Scheme: {scheme}</p>
          <button className="delete" aria-label="close" onClick={handleModal}></button>
        </header>
        <section className="modal-card-body">
          <address>Address: {`${address}, ${city} - ${countryCode}`}</address>
          <p>Last Numbers: {lastNumbers}</p>
          <p>MID: {MID}</p>
        </section>
      </div>
    </div>
  )
} ;

export default Modal;
