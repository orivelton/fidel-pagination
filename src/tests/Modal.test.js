import React from 'react';
import { render } from '@testing-library/react';
import Modal from '../Components/Modal';

describe('component Modal', () => {
  const props = {
    "card": {
      "lastNumbers": "4009",
      "scheme": "visa"
    },
    "location": {
      "address": "Google Street 4",
      "city": "Angular4"
    },
    "identifiers": {
      "MID": "770055002234"
    }
  };

  test('render component Modal', () => {
    const { container } = render(<Modal currentItem={props} />);
    expect(container).not.toBeNull();
    expect(container.firstChild).toMatchSnapshot();
  });
});
