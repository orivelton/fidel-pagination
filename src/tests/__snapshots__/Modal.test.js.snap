// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Modal render component Modal 1`] = `
<div
  class="modal"
>
  <div
    class="modal-background"
  />
  <div
    class="modal-card"
  >
    <header
      class="modal-card-head"
    >
      <p
        class="modal-card-title"
      >
        Scheme: 
        visa
      </p>
      <button
        aria-label="close"
        class="delete"
      />
    </header>
    <section
      class="modal-card-body"
    >
      <address>
        Address: 
        Google Street 4, Angular4 - undefined
      </address>
      <p>
        Last Numbers: 
        4009
      </p>
      <p>
        MID: 
        770055002234
      </p>
    </section>
  </div>
</div>
`;
